packages/webview_flutter/webview_flutter_platform_interface/lib/v4/src/types/navigation_request.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Defines the parameters of the pending navigation callback.
+class NavigationRequest {
+  /// Creates a [NavigationRequest].
+  const NavigationRequest({
+    required this.url,
+    required this.isMainFrame,
+  });
+
+  /// The URL of the pending navigation request.
+  final String url;
+
+  /// Indicates whether to inject the script into the main frame or all frames.
```

Comment:
nit: I think this was copied from another class since this doesnt reference a script

Reply:
I have copied it from Apple's documentation on the `isMainFrame` property. I wasn't sure what it did and looked up where it was used in the native code and then looked up the documentation on Apple's website: https://developer.apple.com/documentation/webkit/wkuserscript/1537856-isformainframeonly

Reply:
Yes, but that documentation has the context of running javascript inside a webpage. This is in context of a `NavigationRequest`, so I think a better description for the context would be `/// Indicates whether the request was made in the web site's main frame or a subframe.`

Reply:
Applied.